File: Backend/Travle.Model/Responses/OrganizerStatsResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// The organizer statistics screen payload (course §2.3): headline counts, revenue and average
    /// rating across the organizer's own tours, the bookings-per-month series scoped to those tours, and
    /// a per-tour breakdown. Scoped server-side to the calling organizer (id from the JWT, never the
    /// request). <see cref="PlatformCommission"/> is bookkeeping only — organizers are never paid out.
    /// </summary>
    public class OrganizerStatsResponse
    {
        public int TotalBookings { get; set; }
        public int PendingBookings { get; set; }
        public int ConfirmedBookings { get; set; }
        public int CompletedBookings { get; set; }
        public int CancelledBookings { get; set; }

        /// <summary>Gross captured across the organizer's tours (before refunds).</summary>
        public decimal GrossRevenue { get; set; }

        /// <summary>Total refunded to travelers across the organizer's tours.</summary>
        public decimal TotalRefunded { get; set; }

        /// <summary>Platform commission on the retained funds (bookkeeping only).</summary>
        public decimal PlatformCommission { get; set; }

        /// <summary>
        /// The organizer's net earnings — gross minus refunds minus platform commission (i.e. their share
        /// of the retained revenue). Notional only: organizers are never actually paid out.
        /// </summary>
        public decimal NetEarnings { get; set; }

        /// <summary>Average of non-removed tour-review ratings across all the organizer's tours.</summary>
        public double AverageRating { get; set; }
        public int ReviewCount { get; set; }

        public string Currency { get; set; } = "bam";

        public List<MonthlyBookingPoint> BookingsPerMonth { get; set; } = new();
        public List<OrganizerTourStatRow> Tours { get; set; } = new();
    }
}

File: UI/travle_core/lib/src/models/organizer_stats_response.dart
import 'package:json_annotation/json_annotation.dart';

import 'monthly_booking_point.dart';
import 'organizer_tour_stat_row.dart';

part 'organizer_stats_response.g.dart';

/// The organizer statistics payload (mirrors the backend `OrganizerStatsResponse`):
/// headline counts, revenue and average rating across the organizer's own tours, the
/// bookings-per-month series and a per-tour breakdown. [platformCommission] is
/// bookkeeping only — organizers are never paid out.
@JsonSerializable()
class OrganizerStatsResponse {
  OrganizerStatsResponse({
    required this.totalBookings,
    required this.pendingBookings,
    required this.confirmedBookings,
    required this.completedBookings,
    required this.cancelledBookings,
    required this.grossRevenue,
    required this.totalRefunded,
    required this.platformCommission,
    required this.netEarnings,
    required this.averageRating,
    required this.reviewCount,
    required this.currency,
    required this.bookingsPerMonth,
    required this.tours,
  });

  final int totalBookings;
  final int pendingBookings;
  final int confirmedBookings;
  final int completedBookings;
  final int cancelledBookings;
  final double grossRevenue;
  final double totalRefunded;
  final double platformCommission;
  final double netEarnings;
  final double averageRating;
  final int reviewCount;
  final String currency;
  final List<MonthlyBookingPoint> bookingsPerMonth;
  final List<OrganizerTourStatRow> tours;

  factory OrganizerStatsResponse.fromJson(Map<String, dynamic> json) =>
      _$OrganizerStatsResponseFromJson(json);

  Map<String, dynamic> toJson() => _$OrganizerStatsResponseToJson(this);
}
